diff -ur nagios.org/html/config.inc.php.in nagios/html/config.inc.php.in
--- nagios.org/html/config.inc.php.in	2013-09-20 21:01:20.000000000 +0200
+++ nagios/html/config.inc.php.in	2013-10-01 21:51:45.176497587 +0200
@@ -2,7 +2,7 @@
 //
 
 
-$cfg['cgi_config_file']='@sysconfdir@/cgi.cfg';  // location of the CGI config file
+$cfg['cgi_config_file']='/etc/webapps/nagios/cgi.cfg';  // location of the CGI config file
 
 $cfg['cgi_base_url']='@cgiurl@';
 
--- nagios.org/Makefile.in	2013-09-20 21:01:20.000000000 +0200
+++ nagios/Makefile.in	2013-10-01 21:51:45.176497587 +0200
@@ -234,14 +234,14 @@ install:
 	cd $(SRC_BASE) && $(MAKE) $@
 	cd $(SRC_CGI) && $(MAKE) $@
 	cd $(SRC_HTM) && $(MAKE) $@
-	$(MAKE) install-exfoliation
+	$(MAKE) install-classicui
 	$(MAKE) install-basic
 
 install-unstripped:
 	cd $(SRC_BASE) && $(MAKE) $@
 	cd $(SRC_CGI) && $(MAKE) $@
 	cd $(SRC_HTM) && $(MAKE) $@
-	$(MAKE) install-exfoliation
+	$(MAKE) install-classicui
 	$(MAKE) install-basic
 
 install-basic:
--- nagios-4.2.1/sample-config/cgi.cfg.in~	2016-09-07 19:22:34.000000000 +0300
+++ nagios-4.2.1/sample-config/cgi.cfg.in	2016-09-07 19:22:52.152175332 +0300
@@ -1,6 +1,6 @@
 #################################################################
 #
-# CGI.CFG - Sample CGI Configuration File for Nagios @VERSION@
+# CGI.CFG - Sample CGI Configuration File for Nagios
 #
 #
 #################################################################
@@ -42,7 +42,7 @@
 # Values: 0 = disables context-sensitive help
 #         1 = enables context-sensitive help
 
-show_context_help=0
+show_context_help=1
 
 
 
@@ -263,7 +263,7 @@
 # OS and distribution, so you may have to tweak this to
 # work on your system.
 
-ping_syntax=/bin/ping -n -U -c 5 $HOSTADDRESS$
+ping_syntax=/usr/bin/ping -n -U -c 5 $HOSTADDRESS$
 
 
 
@@ -402,3 +402,5 @@
 
 #ack_no_sticky=0
 #ack_no_send=0
+
+# vim:ts=8:sw=8:ft=cfg
--- nagios.org/sample-config/nagios.cfg.in	2013-09-20 21:01:20.000000000 +0200
+++ nagios/sample-config/nagios.cfg.in	2013-10-01 21:53:09.708504449 +0200
@@ -1,6 +1,6 @@
 ##############################################################################
 #
-# NAGIOS.CFG - Sample Main Config File for Nagios @VERSION@
+# NAGIOS.CFG - Sample Main Config File for Nagios
 #
 # Read the documentation for more information on this configuration
 # file.  I've provided some comments here, but things may not be so
@@ -15,7 +15,7 @@
 # for historical purposes.  This should be the first option specified 
 # in the config file!!!
 
-log_file=@localstatedir@/nagios.log
+log_file=/var/log/nagios/nagios.log
 
 
 
@@ -26,13 +26,14 @@
 # if you wish (as shown below), or keep them all in a single config file.
 
 # You can specify individual object config files as shown below:
-cfg_file=@sysconfdir@/objects/commands.cfg
-cfg_file=@sysconfdir@/objects/contacts.cfg
-cfg_file=@sysconfdir@/objects/timeperiods.cfg
-cfg_file=@sysconfdir@/objects/templates.cfg
+cfg_file=@sysconfdir@/commands.cfg
+cfg_file=@sysconfdir@/contacts.cfg
+cfg_file=@sysconfdir@/contactgroups.cfg
+cfg_file=@sysconfdir@/timeperiods.cfg
+cfg_file=@sysconfdir@/templates.cfg
 
 # Definitions for monitoring the local (Linux) host
-cfg_file=@sysconfdir@/objects/localhost.cfg
+#cfg_file=@sysconfdir@/objects/localhost.cfg
 
 # Definitions for monitoring a Windows machine
 #cfg_file=@sysconfdir@/objects/windows.cfg
@@ -47,13 +48,8 @@
 # You can also tell Nagios to process all config files (with a .cfg
 # extension) in a particular directory by using the cfg_dir
 # directive as shown below:
-
-#cfg_dir=@sysconfdir@/servers
-#cfg_dir=@sysconfdir@/printers
-#cfg_dir=@sysconfdir@/switches
-#cfg_dir=@sysconfdir@/routers
-
-
+cfg_dir=@sysconfdir@/plugins
+cfg_dir=@sysconfdir@/objects
 
 
 # OBJECT CACHE FILE
@@ -242,7 +238,7 @@
 # This is the directory where archived (rotated) log files should be 
 # placed (assuming you've chosen to do log rotation).
 
-log_archive_path=@localstatedir@/archives
+log_archive_path=/var/log/nagios/archives
 
 
 
@@ -250,7 +246,7 @@
 # If you want messages logged to the syslog facility, as well as the
 # Nagios log file set this option to 1.  If not, set it to 0.
 
-use_syslog=1
+use_syslog=0
 
 
 
@@ -707,7 +703,7 @@
 # by Nagios Enterprises from the update check is processed in accordance 
 # with our privacy policy - see http://api.nagios.org for details.
 
-check_for_updates=1
+check_for_updates=0
 
 
 
@@ -719,7 +715,7 @@
 # this data to determine the number of users running specific version of 
 # Nagios.  Enable this option if you do not want this information to be sent.
 
-bare_update_check=0
+bare_update_check=1
 
 
 
@@ -854,8 +850,8 @@
 # pipe ("p") mode which avoid blocking at startup, otherwise you will
 # likely want the defult append ("a") mode.
 
-#host_perfdata_file_mode=a
-#service_perfdata_file_mode=a
+host_perfdata_file_mode=a
+service_perfdata_file_mode=a
 
 
 
@@ -1091,7 +1087,7 @@
 #	strict-iso8601	(YYYY-MM-DDTHH:MM:SS)
 #
 
-date_format=us
+date_format=iso8601
 
 
 
@@ -1285,7 +1281,7 @@
 # DEBUG FILE
 # This option determines where Nagios should write debugging information.
 
-debug_file=@localstatedir@/nagios.debug
+debug_file=/var/log/nagios/nagios.debug
 
 
 
--- nagios.org/sample-config/resource.cfg.in	2013-09-20 21:01:20.000000000 +0200
+++ nagios/sample-config/resource.cfg.in	2013-10-01 21:52:53.371451113 +0200
@@ -1,6 +1,6 @@
 ###########################################################################
 #
-# RESOURCE.CFG - Sample Resource File for Nagios @VERSION@
+# RESOURCE.CFG - Sample Resource File for Nagios
 #
 #
 # You can define $USERx$ macros in this file, which can in turn be used
