--- ../html/index.html.in~	2005-04-09 23:45:21.000000000 +0300
+++ ../html/index.html.in	2005-04-09 23:45:24.000000000 +0300
@@ -4,6 +4,7 @@
 <HEAD>
 <META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">
 <TITLE>Nagios</TITLE>
+<link rel="shortcut icon" href="@htmurl@/favicon.ico" type="image/ico">
 </HEAD>
 <FRAMESET BORDER="0" FRAMEBORDER="0" FRAMESPACING="0" COLS="180,*">
 <FRAME SRC="side.html" NAME="side" TARGET="main">
--- ../cgi/status.c~	2005-04-09 23:38:44.000000000 +0300
+++ ../cgi/status.c	2005-04-09 23:38:50.000000000 +0300
@@ -499,6 +499,7 @@
 
 	printf("<html>\n");
 	printf("<head>\n");
+	printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 	printf("<title>\n");
 	printf("Current Network Status\n");
 	printf("</title>\n");
--- ../cgi/avail.c~	2005-04-09 23:40:04.000000000 +0300
+++ ../cgi/avail.c	2005-04-09 23:40:06.000000000 +0300
@@ -34,6 +34,7 @@
 
 
 extern char main_config_file[MAX_FILENAME_LENGTH];
+extern char url_html_path[MAX_FILENAME_LENGTH];
 extern char url_images_path[MAX_FILENAME_LENGTH];
 extern char url_stylesheets_path[MAX_FILENAME_LENGTH];
 
@@ -1082,6 +1083,7 @@
 
 	printf("<html>\n");
 	printf("<head>\n");
+	printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 	printf("<title>\n");
 	printf("Nagios Availability\n");
 	printf("</title>\n");
--- ../cgi/cmd.c~	2005-02-02 20:51:20.000000000 +0200
+++ ../cgi/cmd.c	2005-04-09 23:40:44.000000000 +0300
@@ -33,6 +33,7 @@
 #include "../include/getcgi.h"
 
 extern char main_config_file[MAX_FILENAME_LENGTH];
+extern char url_html_path[MAX_FILENAME_LENGTH];
 extern char command_file[MAX_FILENAME_LENGTH];
 extern char comment_file[MAX_FILENAME_LENGTH];
 
@@ -236,6 +237,7 @@
 
 		printf("<html>\n");
 		printf("<head>\n");
+		printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 		printf("<title>\n");
 		printf("External Command Interface\n");
 		printf("</title>\n");
--- ../cgi/config.c~	2004-11-06 07:44:12.000000000 +0200
+++ ../cgi/config.c	2005-04-09 23:41:04.000000000 +0300
@@ -316,6 +316,7 @@
 
 	printf("<html>\n");
 	printf("<head>\n");
+	printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 	printf("<META HTTP-EQUIV='Pragma' CONTENT='no-cache'>\n");
 	printf("<title>\n");
 	printf("Configuration\n");
--- ../cgi/extinfo.c~	2005-02-02 20:51:21.000000000 +0200
+++ ../cgi/extinfo.c	2005-04-09 23:41:26.000000000 +0300
@@ -527,6 +527,7 @@
 
 	printf("<html>\n");
 	printf("<head>\n");
+	printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 	printf("<title>\n");
 	printf("Extended Information\n");
 	printf("</title>\n");
--- ../cgi/histogram.c~	2004-11-06 07:44:12.000000000 +0200
+++ ../cgi/histogram.c	2005-04-09 23:41:53.000000000 +0300
@@ -126,6 +126,7 @@
 
 
 extern char main_config_file[MAX_FILENAME_LENGTH];
+extern char url_html_path[MAX_FILENAME_LENGTH];
 extern char url_images_path[MAX_FILENAME_LENGTH];
 extern char url_stylesheets_path[MAX_FILENAME_LENGTH];
 extern char physical_images_path[MAX_FILENAME_LENGTH];
@@ -997,6 +998,7 @@
 
 		printf("<html>\n");
 		printf("<head>\n");
+		printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 		printf("<title>\n");
 		printf("Nagios Histogram\n");
 		printf("</title>\n");
--- ../cgi/history.c~	2004-12-03 04:01:52.000000000 +0200
+++ ../cgi/history.c	2005-04-09 23:42:21.000000000 +0300
@@ -53,6 +53,7 @@
 int process_cgivars(void);
 
 extern char main_config_file[MAX_FILENAME_LENGTH];
+extern char url_html_path[MAX_FILENAME_LENGTH];
 extern char url_images_path[MAX_FILENAME_LENGTH];
 extern char url_stylesheets_path[MAX_FILENAME_LENGTH];
 
@@ -319,6 +320,7 @@
 
 	printf("<html>\n");
 	printf("<head>\n");
+	printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 	printf("<title>\n");
 	printf("Nagios History\n");
 	printf("</title>\n");
--- ../cgi/notifications.c~	2004-10-31 06:16:02.000000000 +0200
+++ ../cgi/notifications.c	2005-04-09 23:42:42.000000000 +0300
@@ -308,6 +308,7 @@
 
 	printf("<html>\n");
 	printf("<head>\n");
+	printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 	printf("<title>\n");
 	printf("Alert Notifications\n");
 	printf("</title>\n");
--- ../cgi/outages.c~	2004-10-31 06:16:03.000000000 +0200
+++ ../cgi/outages.c	2005-04-09 23:43:00.000000000 +0300
@@ -226,6 +226,7 @@
 
 	printf("<html>\n");
 	printf("<head>\n");
+	printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 	printf("<title>\n");
 	printf("Network Outages\n");
 	printf("</title>\n");
--- ../cgi/statusmap.c~	2004-11-06 07:44:12.000000000 +0200
+++ ../cgi/statusmap.c	2005-04-09 23:43:32.000000000 +0300
@@ -49,6 +49,7 @@
 #define NAGIOS_GD2_ICON       "nagios.gd2"
 
 extern char main_config_file[MAX_FILENAME_LENGTH];
+extern char url_html_path[MAX_FILENAME_LENGTH];
 extern char physical_images_path[MAX_FILENAME_LENGTH];
 extern char url_images_path[MAX_FILENAME_LENGTH];
 extern char url_logo_images_path[MAX_FILENAME_LENGTH];
@@ -326,6 +327,7 @@
 
 		printf("<html>\n");
 		printf("<head>\n");
+		printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 		printf("<title>\n");
 		printf("Network Map\n");
 		printf("</title>\n");
--- ../cgi/summary.c~	2004-11-01 00:23:26.000000000 +0200
+++ ../cgi/summary.c	2005-04-09 23:43:57.000000000 +0300
@@ -34,6 +34,7 @@
 
 
 extern char main_config_file[MAX_FILENAME_LENGTH];
+extern char url_html_path[MAX_FILENAME_LENGTH];
 extern char url_images_path[MAX_FILENAME_LENGTH];
 extern char url_stylesheets_path[MAX_FILENAME_LENGTH];
 
@@ -711,6 +712,7 @@
 
 	printf("<html>\n");
 	printf("<head>\n");
+	printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 	printf("<title>\n");
 	printf("Nagios Event Summary\n");
 	printf("</title>\n");
--- ../cgi/trends.c~	2005-01-17 19:35:34.000000000 +0200
+++ ../cgi/trends.c	2005-04-09 23:44:21.000000000 +0300
@@ -40,6 +40,7 @@
 
 
 extern char main_config_file[MAX_FILENAME_LENGTH];
+extern char url_html_path[MAX_FILENAME_LENGTH];
 extern char url_images_path[MAX_FILENAME_LENGTH];
 extern char url_stylesheets_path[MAX_FILENAME_LENGTH];
 extern char physical_images_path[MAX_FILENAME_LENGTH];
@@ -1115,6 +1116,7 @@
 
 		printf("<html>\n");
 		printf("<head>\n");
+		printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 		printf("<title>\n");
 		printf("Nagios Trends\n");
 		printf("</title>\n");
--- ../cgi/showlog.c~	2004-11-06 07:44:12.000000000 +0200
+++ ../cgi/showlog.c	2005-04-09 23:48:57.000000000 +0300
@@ -33,6 +33,7 @@
 #include "../include/cgiauth.h"
 
 extern char   main_config_file[MAX_FILENAME_LENGTH];
+extern char   url_html_path[MAX_FILENAME_LENGTH];
 extern char   url_images_path[MAX_FILENAME_LENGTH];
 extern char   url_stylesheets_path[MAX_FILENAME_LENGTH];
 
@@ -192,6 +193,7 @@
 
 	printf("<HTML>\n");
 	printf("<HEAD>\n");
+	printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 	printf("<TITLE>\n");
 	printf("Nagios Log File\n");
 	printf("</TITLE>\n");
--- ../cgi/tac.c~	2004-11-06 07:44:12.000000000 +0200
+++ ../cgi/tac.c	2005-04-09 23:52:52.000000000 +0300
@@ -47,6 +47,7 @@
 
 
 extern char   main_config_file[MAX_FILENAME_LENGTH];
+extern char   url_html_path[MAX_FILENAME_LENGTH];
 extern char   url_images_path[MAX_FILENAME_LENGTH];
 extern char   url_stylesheets_path[MAX_FILENAME_LENGTH];
 extern char   url_media_path[MAX_FILENAME_LENGTH];
@@ -330,6 +331,7 @@
 
 	printf("<HTML>\n");
 	printf("<HEAD>\n");
+	printf("<link rel=\"shortcut icon\" href=\"%sfavicon.ico\" type=\"image/ico\">\n", url_html_path);
 	printf("<TITLE>\n");
 	printf("Nagios Tactical Monitoring Overview\n");
 	printf("</TITLE>\n");
