--- nagios-2.0b4/include/locations.h.in~	2003-08-15 04:34:10.000000000 +0300
+++ nagios-2.0b4/include/locations.h.in	2005-11-30 22:23:59.000000000 +0200
@@ -32,7 +32,7 @@
 #define DEFAULT_URL_HTML_PATH			"@htmurl@"
 #define DEFAULT_PHYSICAL_CGIBIN_PATH		"@sbindir@"
 #define DEFAULT_URL_CGIBIN_PATH			"@cgiurl@"
-#define DEFAULT_CGI_CONFIG_FILE			"@sysconfdir@/cgi.cfg"
+#define DEFAULT_CGI_CONFIG_FILE			"/etc/webapps/nagios/cgi.cfg"
 #define DEFAULT_LOCK_FILE			"@lockfile@"
 #define DEFAULT_OBJECT_CACHE_FILE		"@localstatedir@/objects.cache"
 #define DEFAULT_EVENT_BROKER_FILE		"@localstatedir@/broker.socket"
