diff -urNp -x '*.orig' nagios-4.4.6.org/configure.ac nagios-4.4.6/configure.ac
--- nagios-4.4.6.org/configure.ac	2020-04-28 22:48:29.000000000 +0200
+++ nagios-4.4.6/configure.ac	2021-04-19 12:03:10.815220244 +0200
@@ -548,7 +548,7 @@ dnl Should we try and detect the GD libs
 if test "x$TRYGD" = "xyes"; then
 
 	dnl libiconv is required on some systems - tack it on if found
-	AC_CHECK_LIB(iconv,main,ICONV=-liconv,)
+	AC_CHECK_FUNC(iconv,,[AC_CHECK_LIB(iconv,iconv,ICONV=-liconv,)])
 
 	dnl See if the GD lib is available and supports PNG images...
 
