--- sample-config/resource.cfg.in~	2002-02-26 05:04:12.000000000 +0100
+++ sample-config/resource.cfg.in	2004-04-15 13:57:51.065286208 +0200
@@ -21,7 +21,7 @@
 ###########################################################################
 
 # Sets $USER1$ to be the path to the plugins
-$USER1$=@libexecdir@
+$USER1$=@libexecdir@/nagios/plugins
 
 # Sets $USER2$ to be the path to event handlers
 #$USER2$=@libexecdir@/eventhandlers
