diff -ur nagios-4.5.3.org/cgi/extinfo.c nagios-4.5.3/cgi/extinfo.c
--- nagios-4.5.3.org/cgi/extinfo.c	2024-06-11 16:50:20.000000000 +0200
+++ nagios-4.5.3/cgi/extinfo.c	2024-08-01 12:11:08.706637661 +0200
@@ -557,27 +557,6 @@
 		printf("<LINK REL='stylesheet' TYPE='text/css' HREF='%s%s'>\n", url_stylesheets_path, NAGFUNCS_CSS);
 		}
 
-	if (display_type == DISPLAY_HOST_INFO)
-		vidurl = "https://www.youtube.com/embed/n3QEAf-MxY4";
-	else if(display_type == DISPLAY_SERVICE_INFO)
-		vidurl = "https://www.youtube.com/embed/f_knwQOS6FI";
-
-	if (enable_page_tour == TRUE && vidurl) {
-		printf("<script type='text/javascript' src='%s%s'></script>\n", url_js_path, JQUERY_JS);
-		printf("<script type='text/javascript' src='%s%s'></script>\n", url_js_path, NAGFUNCS_JS);
-		printf("<script type='text/javascript'>\n");
-		printf("var vbox, vBoxId='extinfo%d', vboxText = "
-				"'<a href=https://www.nagios.com/tours target=_blank>"
-				"Click here to watch the entire Nagios Core 4 Tour!</a>';\n",
-				display_type);
-		printf("$(document).ready(function() {\n"
-				"var user = '%s';\nvBoxId += ';' + user;\n",
-				current_authdata.username);
-		printf("vbox = new vidbox({pos:'lr',vidurl:'%s',text:vboxText,"
-				"vidid:vBoxId});\n", vidurl);
-		printf("});\n</script>\n");
-		}
-
 	printf("</head>\n");
 
 	printf("<body CLASS='extinfo'>\n");
diff -ur nagios-4.5.3.org/cgi/status.c nagios-4.5.3/cgi/status.c
--- nagios-4.5.3.org/cgi/status.c	2024-06-11 16:50:20.000000000 +0200
+++ nagios-4.5.3/cgi/status.c	2024-08-01 12:11:08.709970995 +0200
@@ -556,31 +556,8 @@
 	/* JS function to append content to elements on page */
 	printf("<script type='text/javascript'>\n");
 	if (enable_page_tour == TRUE) {
-		printf("var vbox, vBoxId='status%d%d', vboxText = "
-				"'<a href=https://www.nagios.com/tours target=_blank>"
-				"Click here to watch the entire Nagios Core 4 Tour!</a>';\n",
-				display_type, group_style_type);
 		printf("$(document).ready(function() {\n"
 				"$('#top_page_numbers').append($('#bottom_page_numbers').html() );\n");
-		if (display_type == DISPLAY_HOSTS)
-			vidurl = "https://www.youtube.com/embed/ahDIJcbSEFM";
-		else if(display_type == DISPLAY_SERVICEGROUPS) {
-			if (group_style_type == STYLE_HOST_DETAIL)
-				vidurl = "https://www.youtube.com/embed/nNiRr0hDZag";
-			else if (group_style_type == STYLE_OVERVIEW)
-				vidurl = "https://www.youtube.com/embed/MyvgTKLyQhA";
-		} else {
-			if (group_style_type == STYLE_OVERVIEW)
-				vidurl = "https://www.youtube.com/embed/jUDrjgEDb2A";
-			else if (group_style_type == STYLE_HOST_DETAIL)
-				vidurl = "https://www.youtube.com/embed/nNiRr0hDZag";
-		}
-		if (vidurl) {
-			printf("var user = '%s';\nvBoxId += ';' + user;",
-				 current_authdata.username);
-			printf("vbox = new vidbox({pos:'lr',vidurl:'%s',text:vboxText,"
-					"vidid:vBoxId});\n", vidurl);
-		}
 		printf("});\n");
 		}
 	printf("function set_limit(url) { \nthis.location = url+'&limit='+$('#limit').val();\n  }\n");
diff -ur nagios-4.5.3.org/cgi/tac.c nagios-4.5.3/cgi/tac.c
--- nagios-4.5.3.org/cgi/tac.c	2024-06-11 16:50:20.000000000 +0200
+++ nagios-4.5.3/cgi/tac.c	2024-08-01 12:11:08.709970995 +0200
@@ -308,17 +308,7 @@
 
 	if (enable_page_tour == TRUE) {
 		printf("<script type='text/javascript' src='%s%s'></script>\n", url_js_path, NAGFUNCS_JS);
-
-		printf("<script type='text/javascript'>\nvar vbox, vBoxId='tac', "
-				"vboxText = '<a href=https://www.nagios.com/tours target=_blank>"
-				"Click here to watch the entire Nagios Core 4 Tour!</a>';\n");
-		printf("$(document).ready(function() {\n"
-				"var user = '%s';\nvBoxId += ';' + user;", current_authdata.username);
-		printf("vbox = new vidbox({pos:'lr',"
-				"vidurl:'https://www.youtube.com/embed/l20YRDhbOfA',text:vboxText,"
-				"vidid:vBoxId});");
-		printf("\n});\n</script>\n");
-		}
+	}
 
 
 
diff -ur nagios-4.5.3.org/html/main.php nagios-4.5.3/html/main.php
--- nagios-4.5.3.org/html/main.php	2024-06-11 16:50:20.000000000 +0200
+++ nagios-4.5.3/html/main.php	2024-08-01 12:21:25.903304305 +0200
@@ -23,21 +23,7 @@ if ($theme != 'dark' && $theme != 'light
 
 <script type='text/javascript'>
 	var cookie;
-	<?php if ($cfg["enable_page_tour"]) { ?>
-		var vbox;
-		var vBoxId = "main";
-		var vboxText = "<a href=https://www.nagios.com/tours target=_blank> " +
-						"Click here to watch the entire Nagios Core 4 Tour!</a>";
-	<?php } ?>
 	$(document).ready(function() {
-		var user = "<?php echo htmlspecialchars($_SERVER['REMOTE_USER'] ?? ''); ?>";
-
-		<?php if ($cfg["enable_page_tour"]) { ?>
-			vBoxId += ";" + user;
-			vbox = new vidbox({pos:'lr',vidurl:'https://www.youtube.com/embed/2hVBAet-XpY',
-								text:vboxText,vidid:vBoxId});
-		<?php } ?>
-
 		getCoreStatus();
 	});
 
@@ -81,12 +67,11 @@ if ($theme != 'dark' && $theme != 'light
 
 <div id="updateversioninfo">
 <?php
-	$updateinfo = get_update_information();
-	if (!$updateinfo['update_checks_enabled']) {
+	if (1 || !$updateinfo['update_checks_enabled']) {
 ?>
 		<div class="updatechecksdisabled">
 			<div class="warningmessage">Warning: Automatic Update Checks are Disabled!</div>
-			<div class="submessage">Disabling update checks presents a possible security risk.  Visit <a href="https://www.nagios.org/" target="_blank">nagios.org</a> to check for updates manually or enable update checks in your Nagios config file.</a></div>
+			<div class="submessage">Disabling update checks presents a possible security risk.  Visit <a href="https://www.nagios.org/" target="_blank">nagios.org</a> to check for updates manually.</a></div>
 		</div>
 <?php
 	} else if ($updateinfo['update_available'] && $this_version < $updateinfo['update_version']) {
