--- nagios-3.4.1/html/main.php~	2012-05-12 03:00:53.000000000 +0300
+++ nagios-3.4.1/html/main.php	2012-05-28 23:55:54.310721306 +0300
@@ -14,7 +14,7 @@ $this_year = '2017';
 <title>Nagios Core</title>
 <link rel="stylesheet" type="text/css" href="stylesheets/common.css?<?php echo $this_version; ?>" />
 <link rel="stylesheet" type="text/css" href="stylesheets/nag_funcs.css?<?php echo $this_version; ?>" />
-<script type="text/javascript" src="js/jquery-1.12.4.min.js"></script>
+<script type="text/javascript" src="/js/jquery/jquery.js"></script>
 <script type="text/javascript" src="js/nag_funcs.js"></script>
 
 <script type='text/javascript'>
--- nagios-3.4.1/include/cgiutils.h~	2012-05-24 17:15:30.000000000 +0300
+++ nagios-3.4.1/include/cgiutils.h	2012-05-28 23:56:18.921757228 +0300
@@ -93,7 +93,7 @@ NAGIOS_BEGIN_DECL
 #define NAGFUNCS_CSS       "nag_funcs.css"
 
 	/********************************* JAVASCRIPT INCLUDES **********************/
-#define JQUERY_JS          "jquery-1.12.4.min.js"
+#define JQUERY_JS          "/js/jquery/jquery.js"
 #define NAGFUNCS_JS        "nag_funcs.js"
 
 	/********************************* ICONS ************************************/
--- nagios-3.4.1/cgi/status.c~	2012-05-24 17:15:30.000000000 +0300
+++ nagios-3.4.1/cgi/status.c	2012-05-28 23:57:04.933694000 +0300
@@ -531,7 +531,7 @@ void document_header(int use_stylesheet)
 		}
 
 	/* added jquery library 1/31/2012 */
-	printf("<script type='text/javascript' src='%s%s'></script>\n", url_js_path, JQUERY_JS);
+	printf("<script type='text/javascript' src='%s'></script>\n",JQUERY_JS);
 	printf("<script type='text/javascript' src='%s%s'></script>\n", url_js_path, NAGFUNCS_JS);
 	/* JS function to append content to elements on page */
 	printf("<script type='text/javascript'>\n");
--- nagios-3.4.1/html/Makefile.in~	2012-05-28 23:52:30.000000000 +0300
+++ nagios-3.4.1/html/Makefile.in	2012-05-28 23:59:43.973722231 +0300
@@ -61,8 +61,6 @@
 	do $(INSTALL) -m 664 $(INSTALL_OPTS) $$file $(DESTDIR)$(HTMLDIR)/stylesheets; done
 	for file in contexthelp/*.html; \
 	do $(INSTALL) -m 664 $(INSTALL_OPTS) $$file $(DESTDIR)$(HTMLDIR)/contexthelp; done
-	for file in js/*.js; \
-	do $(INSTALL) -m 664 $(INSTALL_OPTS) $$file $(DESTDIR)$(HTMLDIR)/js; done
 #	for file in docs/*.html; \
 #	do $(INSTALL) -m 664 $(INSTALL_OPTS) $$file $(DESTDIR)$(HTMLDIR)/docs; done
 #	for file in docs/images/*.*; \
